app/assets/javascripts/covid19/src/webapp/components/structures-table/StructuresTable.tsx

```diff
@@ -90,22 +96,39 @@ export const StructuresTable: React.FC<StructuresTableProps> = React.memo(props
     const { structures } = filteredData;
 
     const showDetailsDialog = React.useCallback(
-        (options: { row: Structure; field: Field }) => {
+        (options: ViewMoreOptions) => {
             sendAnalytics({
                 type: "event",
                 action: "open",
                 category: "dialog",
                 label: `Details. Field: ${options.field}, PDB: ${options.row.pdb?.id}`,
             });
-            openDialog();
+            openViewMore();
             setDetailsOptions({ field: options.field, structure: options.row });
         },
-        [openDialog]
+        [openViewMore]
+    );
+
+    const showIDRDialog = React.useCallback(
```

Comment:
This component is growing a bit too much. Let's use custom hooks for any code we add/refactor.

Reply:
I joined both details and idr states and callbacks into useInfoDialog<T> hook.